test_util/aws.py

```diff
@@ -108,22 +129,116 @@ def delete(self):
         bucket.delete()
         logger.info('Delete successfully triggered for {}'.format(self.stack.stack_name))
 
-    def get_tag_instances(self, tag):
-        return self.boto_wrapper.client('ec2').describe_instances(
-            Filters=[
-                {'Name': 'tag-value', 'Values': [self.stack.stack_name]},
-                {'Name': 'tag-value', 'Values': [tag]}])['Reservations'][0]['Instances']
+
+class DcosCfAdvanced(CfStack):
+    @classmethod
+    def create(cls, stack_name, boto_wrapper, template_url,
+               public_agents, private_agents, key_pair_name,
+               private_agent_type, public_agent_type, master_type,
+               vpc_cidr='10.0.0.0/16', public_subnet_cidr='10.0.128.0/20',
+               private_subnet_cidr='10.0.0.0/17',
+               gateway=None, vpc=None, private_subnet=None, public_subnet=None):
+        ec2 = boto_wrapper.client('ec2')
+        if not vpc:
+            logging.info('Creating new VPC...')
+            vpc = ec2.create_vpc(CidrBlock=vpc_cidr, InstanceTenancy='default')['Vpc']['VpcId']
+            ec2.get_waiter('vpc_available').wait(VpcIds=[vpc])
+            ec2.create_tags(Resources=[vpc], Tags=[{'Key': 'Name', 'Value': stack_name}])
+        logging.info('Using VPC with ID: ' + vpc)
+
+        if not gateway:
```

Comment:
I agree that leaving this in is dubious without having a better cleanup method, but I feel leaving it in is a convenient addition (e.g. it took just a second for me to make the shared Vpc, gateway, subnet combo for eu-central-1). That said, its pretty trivial so if you insist on taking it out, I wont argue